### PR TITLE
rhui(gcp): add SAP RHEL 8.10 client

### PR DESCRIPTION
**Pull request by MichalHe**
Add a dedicated entry for SAP RHEL 8.10 on GCP due to the same reasons as with other cloud providers - 8.10 is the last RHEL8 release, thus, there are no content channels. A dedicated client is introduced, providing SAP systems with GA content.

### CODE REVIEW

**Comment by alexxa** [comment]
```
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Erasing          : google-rhui-client-rhel79-sap-4.0-1.noarch             1/1 
  Verifying        : google-rhui-client-rhel79-sap-4.0-1.noarch             1/1 

Removed:
  google-rhui-client-rhel79-sap-4.0-1.noarch                                    

Last metadata expiration check: 0:09:48 ago on Mon Jun 10 09:57:09 2024.
> Leaving Shell
Process Process-546:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/repository/actor_definition.py", line 74, in _do_run
    actor_instance.run(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/actors/__init__.py", line 289, in run
    self.process(*args)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/actor.py", line 58, in process
    userspacegen.perform()
  File "/usr/lib/python2.7/site-packages/leapp/utils/deprecation.py", line 42, in process_wrapper
    return target_item(*args, **kwargs)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 1252, in perform
    setup_target_rhui_access_if_needed(context, indata)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 1223, in setup_target_rhui_access_if_needed
    files_owned_by_clients = _query_rpm_for_pkg_files(context, indata.rhui_info.target_client_pkg_names)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 288, in _query_rpm_for_pkg_files
    rpm_query_result = context.call(['rpm', '-ql'] + pkgs, split=True)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/mounting.py", line 168, in call
    return run(self.type.make_command(cmd), *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/libraries/stdlib/__init__.py", line 192, in run
    result=result
CalledProcessError: Command ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=saphana', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=869f7048-df87-4b35-9f43-2586e5fc2a7f', '--setenv=LEAPP_HOSTNAME=instance-20240610-093729.c.sap-qe.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap'] failed with exit code 1.
2024-06-10 10:06:57.937 ERROR    PID: 11125 leapp.workflow.TargetTransactionFactsCollection: Actor target_userspace_creator has crashed: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/leapp/repository/actor_definition.py", line 74, in _do_run
    actor_instance.run(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/actors/__init__.py", line 289, in run
    self.process(*args)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/actor.py", line 58, in process
    userspacegen.perform()
  File "/usr/lib/python2.7/site-packages/leapp/utils/deprecation.py", line 42, in process_wrapper
    return target_item(*args, **kwargs)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 1252, in perform
    setup_target_rhui_access_if_needed(context, indata)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 1223, in setup_target_rhui_access_if_needed
    files_owned_by_clients = _query_rpm_for_pkg_files(context, indata.rhui_info.target_client_pkg_names)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py", line 288, in _query_rpm_for_pkg_files
    rpm_query_result = context.call(['rpm', '-ql'] + pkgs, split=True)
  File "/usr/share/leapp-repository/repositories/system_upgrade/common/libraries/mounting.py", line 168, in call
    return run(self.type.make_command(cmd), *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/libraries/stdlib/__init__.py", line 192, in run
    result=result
CalledProcessError: Command ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=saphana', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=869f7048-df87-4b35-9f43-2586e5fc2a7f', '--setenv=LEAPP_HOSTNAME=instance-20240610-093729.c.sap-qe.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap'] failed with exit code 1.



=========================================================================================================
Actor target_userspace_creator unexpectedly terminated with exit code: 1 - Please check the above details
=========================================================================================================

Debug output written to /var/log/leapp/leapp-preupgrade.log


```

File: repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py [hunk 1]
```diff
@@ -1220,7 +1220,20 @@ def setup_target_rhui_access_if_needed(context, indata):
     _apply_rhui_access_postinstall_tasks(context, setup_info)
 
     # Do a cleanup so there are not duplicit repoids
-    files_owned_by_clients = _query_rpm_for_pkg_files(context, indata.rhui_info.target_client_pkg_names)
+    try:
+        files_owned_by_clients = _query_rpm_for_pkg_files(context, indata.rhui_info.target_client_pkg_names)
+    except CalledProcessError as err:  # We failed to rpm -qf PKG, the PKG is most likely not installed
+        api.current_logger().critical('Failed to query files owned by target RHUI clients (clients=%s). This is caused'
+                                      ' by failing to install the target clients during the client-swap step.'
+                                      'Full error: %s', indata.rhui_info.target_client_pkg_names, err)
```
Comment (abadger) [inline repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py:1228]:
```suggestion
                                      ' Full error: %s', indata.rhui_info.target_client_pkg_names, err)
```

File: repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py [hunk 2]
```diff
@@ -1220,7 +1220,20 @@ def setup_target_rhui_access_if_needed(context, indata):
...
+
+        target_major = get_target_major_version()
+        plural_suffix = 's' if len(indata.rhui_info.target_client_pkg_names) > 1 else ''
+        client_rpms = ', '.join(indata.rhui_info.target_client_pkg_names)
+        msg = ('Could not find the RHEL {target_major} RHUI client rpm{plural_suffix} ({client_rpms})'
+               'in the cloud provider\'s client repository.')
```
Comment (abadger) [inline repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py:1234]:
```suggestion
               ' in the cloud provider\'s client repository.')
```

**Comment by Rezney** [comment]
I took the `google-rhui-client-rhel810-sap` package we were provided, created a custom repo to use it and got:

```
2024-06-10 18:18:52.357 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has started: ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'dnf', '-y', '--disablerepo', '*', '--enablerepo', 'google-compute-engine-leappx', '--enablerepo', 'google-compute-engine-leapp', '--setopt=module_platform_id=platform:el8', '--setopt=keepcache=1', '--releasever', u'8.10', '--disableplugin', 'subscription-manager', 'shell']
Host and machine ids are equal (24188c16689e67352a79d8ba9f4395bc): refusing to link journals
Google Compute Engines                          0.0  B/s |   0  B     00:00    
Google Compute Engine                           0.0  B/s |   0  B     00:00    
> > > ================================================================================
 Package                        Arch   Version
                                             Repository                    Size
================================================================================
Installing:
Package google-rhui-client-rhel810-sap-4.4-1.noarch.rpm is not signed
 google-rhui-client-rhel810-sap noarch 4.4-1 google-compute-engine-leappx  15 k
Removing:
 google-rhui-client-rhel79-sap  noarch 4.0-1 @System                       27 k

Transaction Summary
================================================================================
Install  1 Package
Remove   1 Package

Total size: 15 k
Downloading Packages:
Last metadata expiration check: 0:00:01 ago on Mon Jun 10 18:10:37 2024.
> Leaving Shell
2024-06-10 18:18:53.209 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has finished: ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'dnf', '-y', '--disablerepo', '*', '--enablerepo', 'google-compute-engine-leappx', '--enablerepo', 'google-compute-engine-leapp', '--setopt=module_platform_id=platform:el8', '--setopt=keepcache=1', '--releasever', u'8.10', '--disableplugin', 'subscription-manager', 'shell']
2024-06-10 18:18:53.211 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Applying RHUI postinstall tasks.
2024-06-10 18:18:53.212 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has started: ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap']
Host and machine ids are equal (24188c16689e67352a79d8ba9f4395bc): refusing to link journals
package google-rhui-client-rhel810-sap is not installed
2024-06-10 18:18:53.274 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Command ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap'] failed with exit code 1.
2024-06-10 18:18:53.279 DEBUG    PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has finished: ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap']
2024-06-10 18:18:53.281 CRITICAL PID: 13039 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: Failed to query files owned by target RHUI clients (clients=[u'google-rhui-client-rhel810-sap']). This is caused by failing to install the target clients during the client-swap step. Full error: Command ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.10', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=ga', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_ENABLE_REPOS=google-compute-engine-leappx', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=84e9f548-38ea-4ba2-88e4-48b67e8ae6d8', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'rpm', '-ql', u'google-rhui-client-rhel810-sap'] failed with exit code 1.
```

It looks like we can find it, install it, but later we cannot see it when running `rpm -ql google-rhui-client-rhel810-sap`

In the transaction, it is printing `Package google-rhui-client-rhel810-sap-4.4-1.noarch.rpm is not signed` but it should not matter in this particular phase. 

@MichalHe WDYT? ^

**Comment by Rezney** [comment]
Mmm, this is the dnf shell outputwhen upgrading to 8.8:

```
2024-06-10 19:02:54.351 DEBUG    PID: 25705 leapp.workflow.TargetTransactionFactsCollection.target_userspace_creator: External command has started: ['systemd-nspawn', '--register=no', '--quiet', '-D', '/var/lib/leapp/scratch/mounts/root_/system_overlay', '--setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.8', '--setenv=LEAPP_NO_RHSM=1', '--setenv=LEAPP_EXPERIMENTAL=0', '--setenv=LEAPP_TARGET_PRODUCT_CHANNEL=e4s', '--setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', '--setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade/common/tools:/etc/leapp/repos.d/system_upgrade/el7toel8/tools', '--setenv=LEAPP_NO_RHSM_FACTS=1', '--setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade/common/files:/etc/leapp/repos.d/system_upgrade/el7toel8/files', '--setenv=LEAPP_IPU_IN_PROGRESS=7to8', '--setenv=LEAPP_UNSUPPORTED=0', '--setenv=LEAPP_EXECUTION_ID=ea21cc57-fa7c-44de-a778-05463444353d', '--setenv=LEAPP_HOSTNAME=mreznik-sap.c.leapp-devel-700.internal', 'dnf', '-y', '--disablerepo', '*', '--enablerepo', 'google-compute-engine-leappx', '--enablerepo', 'google-compute-engine-leapp', '--setopt=module_platform_id=platform:el8', '--setopt=keepcache=1', '--releasever', u'8.8', '--disableplugin', 'subscription-manager', 'shell']
Host and machine ids are equal (24188c16689e67352a79d8ba9f4395bc): refusing to link journals
Last metadata expiration check: 0:13:27 ago on Mon Jun 10 18:49:27 2024.
> > > ================================================================================
 Package                       Arch   Version Repository                   Size
================================================================================
Installing:
 google-rhui-client-rhel8-sap  noarch 4.0-1   google-compute-engine-leapp  15 k
Removing:
 google-rhui-client-rhel79-sap noarch 4.0-1   @System                      27 k

Transaction Summary
================================================================================
Install  1 Package
Remove   1 Package

Total size: 15 k
Downloading Packages:
[SKIPPED] b1f0f31f84a08378afeb755f0c8396e1b8f84e066a3983db7aca63897ba4ed26-google-rhui-client-rhel8-sap-4.0-1.noarch.rpm: Already downloaded                         
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : google-rhui-client-rhel8-sap-4.0-1.noarch              1/2 
  Running scriptlet: google-rhui-client-rhel8-sap-4.0-1.noarch              1/2 
  Erasing          : google-rhui-client-rhel79-sap-4.0-1.noarch             2/2 
  Verifying        : google-rhui-client-rhel8-sap-4.0-1.noarch              1/2 
  Verifying        : google-rhui-client-rhel79-sap-4.0-1.noarch             2/2 

Installed:
  google-rhui-client-rhel8-sap-4.0-1.noarch                                     

Removed:
  google-rhui-client-rhel79-sap-4.0-1.noarch                                    

Last metadata expiration check: 0:13:29 ago on Mon Jun 10 18:49:27 2024.
> Leaving Shell
```

There may be something wrong with my approach of adding the custom repo. Checking further.

**Comment by Rezney** [comment]
Ok, sorry for the fuss, I did not use `gpgcheck=0` in my custom repo. Upgrading RHEL for SAP 7.9 to 8.10 is now passing. Approving the PR.

**Review by Rezney — APPROVED** [review]

**Pull request merged**